apps/playground/src/pages/yjs-quill-wio.tsx

```diff
@@ -0,0 +1,207 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import ReactQuill, { Quill } from "react-quill-new";
+import { QuillBinding } from "y-quill";
+import "react-quill-new/dist/quill.snow.css";
+import QuillCursors from "quill-cursors";
+import { getConfig } from "../config";
+
+Quill.register("modules/cursors", QuillCursors);
+
+const SUPERVIZ_KEY = getConfig<string>("keys.superviz");
+const SUPERVIZ_ROOM_PREFIX = getConfig<string>("roomPrefix");
+
+const componentName = "yjs-quill-wio";
+
+function setStyles(
+  states: Map<number, Record<string, any>>,
+  ids: Set<number>
+): number[] {
+  const stylesheet = document.getElementById("sv-yjs-quill");
+  let styles = "";
+
+  const idsList = [];
+  for (const [id, state] of states) {
+    if (ids.has(id) || !state.participant) continue;
+    idsList.push(id);
+
+    styles += `
+        #ql-cursor-${id} {
+          --presence-color: ${state.participant.slot.color};
+          --sv-text-color: ${state.participant.slot.textColor};
+      }
+    `;
+  }
+
+  stylesheet!.innerText = styles;
+
+  return idsList;
+}
+
+export function YjsQuillWio() {
+  const ydoc = useMemo(() => new Y.Doc(), []);
+
+  const [localParticipant, setLocalParticipant] =
+    useState<Partial<Participant>>();
+  const [ids, setIds] = useState(new Set<number>());
+  const [joinedRoom, setJoinedRoom] = useState(false);
+
+  const room = useRef<LauncherFacade>();
+  const quillRef = useRef<ReactQuill | null>(null);
+  const provider = useMemo<SuperVizYjsProvider>(
+    () => new SuperVizYjsProvider(ydoc),
+    [ydoc]
+  );
+  const wio = useRef<WhoIsOnline>();
+  const loaded = useRef(false);
+
+  const initializeSuperViz = useCallback(async () => {
+    if (loaded.current) return;
+    loaded.current = true;
+
+    const uuid = generateId();
+
+    room.current = await Room(SUPERVIZ_KEY, {
+      roomId: `${SUPERVIZ_ROOM_PREFIX}-${componentName}`,
+      participant: {
+        name: "Participant",
+        id: uuid,
+      },
+      group: {
+        name: SUPERVIZ_ROOM_PREFIX,
+        id: SUPERVIZ_ROOM_PREFIX,
+      },
+      environment: "dev",
+      debug: true,
+    });
+
+    wio.current = new WhoIsOnline();
+
+    room.current.subscribe("participant.updated", (data) => {
+      if (!data.slot?.index) return;
+
+      provider.awareness?.setLocalStateField("participant", {
+        id: data.id,
+        slot: data.slot,
+        name: data.name,
+      });
+
+      setLocalParticipant({
+        id: data.id,
+        slot: data.slot,
+        name: data.name,
+      });
+    });
+
+    const style = document.createElement("style");
+    style.id = "sv-yjs-quill";
+    document.head.appendChild(style);
+  }, [provider.awareness]);
+
+  useEffect(() => {
+    initializeSuperViz();
+
+    return () => {
+      room.current?.removeComponent(wio.current);
+      room.current?.removeComponent(provider);
+      room.current?.destroy();
+    };
+  }, [initializeSuperViz, provider]);
```

Comment:
The dependency array must be empty to destroy only when the component is unmounted.


```suggestion
  useEffect(() => {
    initializeSuperViz();

    return () => {
      room.current?.removeComponent(wio.current);
      room.current?.removeComponent(provider);
      room.current?.destroy();
    };
  }, []);
```